File: src/main/java/Pages/HomePage.java
package Pages;

import Base.TestUtilities;
import org.apache.logging.log4j.Logger;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.testng.Assert;
import org.testng.annotations.Test;

public class HomePage extends BasePageObject {

    private String pageUrl = "http://automationpractice.com/index.php";

    private By phoneNumberLocator = By.cssSelector("nav  strong");
    private By searchBarLocator = By.xpath("//input[@id='search_query_top']");
    private By searchButtonLocator = By.xpath("//button[@name='submit_search']");
    private By shortSleeveLocator = By.xpath("//h5/a[@title='Faded Short Sleeve T-shirts']");

    public HomePage(WebDriver driver, Logger log) {

        super(driver, log);
    }

    /**
     * Open HomePage with it's url
     */
    public void openHomePage() {
        log.info("Opening page " + pageUrl);
        openPage(pageUrl);
        log.info(pageUrl + " opened!");
    }


    public String getPhoneNumberText() {
        String phoneNumberText = findElement(phoneNumberLocator).getText();

        return phoneNumberText;
    }




}

File: src/main/java/pages/HomePage.kt
package pages

import base.BasePageObject

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.support.PageFactory

class HomePage(driver: WebDriver) : BasePageObject(driver)  {

    fun openHomePage() {
       openPage(HOME_PAGE_URL)
    }

    fun clickOnItemCard(): ItemDetailPage {
        clickOnElement(shortSleeveLocator)

        return ItemDetailPage(driver)
    }

    fun clickOnContactUs(): ContactUsPage {
        clickOnElement(contactUs)

        return ContactUsPage(driver)
    }

    fun clickOnSignIn(): SignInPage {
        clickOnElement(signInLocator)

        return SignInPage(driver)
    }

    fun getPhoneNumberText(): String {
       return findElement(phoneNumberLocator).text
    }

    init {
        PageFactory.initElements(driver, this)
    }

    companion object {
        private const val HOME_PAGE_URL = "http://automationpractice.com/index.php"

        private val phoneNumberLocator = By.cssSelector("nav  strong")
        private val shortSleeveLocator = By.xpath("//h5/a[@title='Faded Short Sleeve T-shirts']")
        private val contactUs = By.xpath("//*[@title='Contact Us']")
        private val signInLocator = By.xpath("//*[@title='Log in to your customer account']")

    }
}